backend/app/api/endpoints/billing.py

```
@@ -89,3 +89,48 @@ async def create_checkout_session(
+            customer=customer_id,
+            return_url=f"{frontend_base}/dashboard/settings?tab=billing",
+        )
+    except stripe.StripeError:
+        logger.error(
+            "Stripe portal session failed for user %s",
+            current_user.get("auth_id"),
+            exc_info=True,
+        )
+        raise HTTPException(
+            status_code=502, detail="Payment provider error — please try again"
+        )
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)